### PR TITLE
phat updates to cylon_yolol.md

### PR DESCRIPTION
**Pull request by ocornoc**
Firstly: fixed the precision errors (back when it was understood that Yolol numbers had 4 rather than 3 decimals of precision)
Secondly: Mentioned that it's undefined to increment or decrement a temporary value or constant
Finally: added a section describing much of multichip and network semantics in Yolol

### CODE REVIEW

**Comment by ocornoc** [comment]
Once it's known what setting `:ChipWait` to a non-integer number or string does, I'm happy to open for merging.

**Comment by Azurethi** [comment]
If I understand Artturi correctly, non-ints still just get one subtracted until they enter the interval [0,1) whereas strings are ignored & the chips continue operating. @ocornoc

> ArtturiFB: It subtracts 1 from the value every cycle, and executes when 0<=chipWait<1
> ArtturiFB: With strings, it can't subtract 1 so the chip isn't paused

See these messages on Cylon: https://discordapp.com/channels/590281942184755239/596028165042733081/677022606062256138

File: information/cylon_yolol.md [hunk 1]
```diff
@@ -90,3 +90,15 @@ Basic (`a = b`) and compound (`a += b`, etc.) assignment operators are a special
  * `a *= b`
  * `a /= b`
  * `a %= b`
+
+It is undefined to increment/decrement a temporary value or constant.
+
+## Multichip and Network Semantics
+
+Yolol network semantics are complicated. For every network, there are at least zero chips. For each chip, there are 20 lines. Chips push a "line run event" once every tick (assuming their `:chipwait` field is set to `0`; what this field does will be explained further). Networks flush their device queue once every tick as well. When a line run event is popped from the network device queue, the chip that pushed it has that line executed and the network's state (including chips' local variables) is updated in accordance with this line execution. This, in fact, makes lines atomic with respect to a network.
```
Comment (martindevans) [inline information/cylon_yolol.md]: Opening with `X is complicated` in a section explaining X seems a bit... odd. Maybe instead open by defining what a network actually is. i.e. a set of chips/devices all physically wired together, sharing a common set of global fields on devices.

File: information/cylon_yolol.md [hunk 2]
```diff
@@ -90,3 +90,15 @@ Basic (`a = b`) and compound (`a += b`, etc.) assignment operators are a special
...
+
+This can cause some nondeterminism. How do I know Chip A's line will execute before or after Chip B's? Will it cause a data race? Will the world implode? Likely not, but these are interesting questions. Whether a chip will go before or after another chip when executing its line run event is known to be nondeterministic, and developers have very nearly confirmed that it's not necessarily true that the ordering will be preserved across network queue flushes.
```
Comment (martindevans) [inline information/cylon_yolol.md]: This document is for answering questions, not asking them xD

Comment (ocornoc) [reply information/cylon_yolol.md]: lol

File: information/cylon_yolol.md [hunk 3]
```diff
@@ -90,3 +90,15 @@ Basic (`a = b`) and compound (`a += b`, etc.) assignment operators are a special
  * `a *= b`
  * `a /= b`
  * `a %= b`
+
+It is undefined to increment/decrement a temporary value or constant.
```
[the remark this answers is not in the document]
Comment (ocornoc) [reply information/cylon_yolol.md]: You're right, still using my lizard C/++ brain.

File: information/cylon_yolol.md [hunk 4]
```diff
@@ -41,7 +41,7 @@ Syntax errors result in the _line_ (not statement) being parsed to be completely
 ## Semantics
 
  * The factorial operator (`!`) is guaranteed to be equal to the factorial function at nonnegative integers. Other inputs are either a runtime error if not supported, or may return a number using another function (Gamma, flooring, etc.).
- * The maximum value for a number is `922337203685477.5807` (MAX_VALUE) and the minimum value `-922337203685477.5808` (MIN_VALUE). Only four decimal digits of sub-integer precision are guaranteed.
+ * The maximum value for a number is `9223372036854775.807` ("MAX_VALUE" here) and the minimum value `-9223372036854775.808` ("MIN_VALUE" here). Only three decimal digits of sub-integer precision are guaranteed.
```
[the remark this answers is not in the document]
Comment (ocornoc) [reply information/cylon_yolol.md]: I'll reword it to that.
